pkg/backup/util/util.go

```diff
@@ -86,24 +86,40 @@ func GenerateS3CertEnvVar(s3 *v1alpha1.S3StorageProvider) ([]corev1.EnvVar, stri
 			Name:  "AWS_STORAGE_CLASS",
 			Value: s3.StorageClass,
 		},
-		{
-			Name: "AWS_ACCESS_KEY_ID",
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{Name: s3.SecretName},
-					Key:                  constants.S3AccessKey,
+	}
+	accessKeyName := "AWS_ACCESS_KEY_ID"
+	secretKeyName := "AWS_SECRET_ACCESS_KEY"
+	if useKMS {
+		envVars = append(envVars, []corev1.EnvVar{
+			{
+				Name:  "AWS_DEFAULT_REGION",
+				Value: s3.Region,
+			},
+		}...)
+		accessKeyName = fmt.Sprintf("%s_AWS_ACCESS_KEY_ID", constants.KMSSecretPrefix)
+		secretKeyName = fmt.Sprintf("%s_AWS_SECRET_ACCESS_KEY", constants.KMSSecretPrefix)
+	}
+	if s3.SecretName != "" {
```

Comment:
We should put `if s3.SecretName == ""{return  envVars, "", nil}` on line 90 to avoid doing useless work.

Reply:
If we do like this, the KMS env "AWS_DEFAULT_REGION" may be miss setting.